src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -456,97 +444,105 @@ private static void PerformInjection()
                 return;
             }
 
-            if (this.IsEnabled(ActivityName, request))
-            {
-                // We don't call StartActivity here because it will fire into user code before the headers are added.
+            var activity = ActivitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                Activity.Current?.Context ?? default,
+                this.BuildRequestTags(request));
 
-                var activity = new Activity(ActivityName);
-                activity.Start();
+            Debug.Assert(activity != null, "Activity null check failed.");
 
-                IAsyncResult asyncContext = readAResultAccessor(request);
-                if (asyncContext != null)
-                {
-                    // Flow here is for [Begin]GetResponse[Async] without a prior call to [Begin]GetRequestStream[Async].
+            IAsyncResult asyncContext = readAResultAccessor(request);
+            if (asyncContext != null)
+            {
+                // Flow here is for [Begin]GetResponse[Async] without a prior call to [Begin]GetRequestStream[Async].
 
-                    AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
-                    asyncCallbackModifier(asyncContext, callback.AsyncCallback);
-                }
-                else
-                {
-                    // Flow here is for [Begin]GetRequestStream[Async].
+                AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
+                asyncCallbackModifier(asyncContext, callback.AsyncCallback);
+            }
+            else
+            {
+                // Flow here is for [Begin]GetRequestStream[Async].
 
-                    asyncContext = writeAResultAccessor(request);
-                    AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
-                    asyncCallbackModifier(asyncContext, callback.AsyncCallback);
-                }
+                asyncContext = writeAResultAccessor(request);
+                AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
+                asyncCallbackModifier(asyncContext, callback.AsyncCallback);
+            }
 
-                InstrumentRequest(request, activity);
+            InstrumentRequest(request, activity);
 
-                // Only send start event to users who subscribed for it, but start activity anyway
-                if (this.IsEnabled(RequestStartName))
-                {
-                    this.Write(activity.OperationName + ".Start", new { Request = request });
-                }
-            }
+            activity.SetCustomProperty("request", request);
```

Comment:
Did our reflection hack DiagnosticSource inject w3ctraceparent headers? Or was it done at Listener?
In the new model, listener won't be able to do it, as "request" is part of customProperty, and not available at StartActivity callback.